contribs/gnodev/setup_node.go

```diff
@@ -12,9 +12,45 @@ import (
 	"github.com/gnolang/gno/contribs/gnodev/pkg/emitter"
 	"github.com/gnolang/gno/contribs/gnodev/pkg/packages"
 	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	"github.com/gnolang/gno/gno.land/pkg/gnoland/ugnot"
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
 	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/std"
 )
 
+// extractDependenciesFromTxs extracts dependencies from transactions and adds them to the paths slice and config.BalancesList.
+func extractDependenciesFromTxs(nodeConfig *gnodev.NodeConfig, paths *[]string) {
+	for _, tx := range nodeConfig.InitialTxs {
+		for _, msg := range tx.Tx.Msgs {
+			// TODO: Support MsgRun
+			if callMsg, ok := msg.(vm.MsgCall); ok {
+				// Add package path to paths slice if not already present
+				if !slices.Contains(*paths, callMsg.PkgPath) {
+					*paths = append(*paths, callMsg.PkgPath)
+				}
+
+				// Check if address exists in config.BalancesList
+				addressExists := false
+				for _, balance := range nodeConfig.BalancesList {
+					if balance.Address == callMsg.Caller {
+						addressExists = true
+						break
+					}
+				}
+
+				// If address does not exist, add it to config.BalancesList
+				if !addressExists {
+					newBalance := gnoland.Balance{
+						Address: callMsg.Caller,
+						Amount:  std.Coins{std.NewCoin(ugnot.Denom, 10e12)},
```

Comment:
Can you centralize this in a single top-level variable? Set the one in accounts as the top-level variable  
`var defaultPremineBalance = std.Coins{std.NewCoin(ugnot.Denom, 10e12)}`  
to have a single common reference.